pkg/scheduler/framework/plugins/tainttoleration/taint_toleration.go

```diff
@@ -167,3 +170,23 @@ func (pl *TaintToleration) ScoreExtensions() framework.ScoreExtensions {
 func New(_ context.Context, _ runtime.Object, h framework.Handle) (framework.Plugin, error) {
 	return &TaintToleration{handle: h}, nil
 }
+
+// isSchedulableAfterPodChange is invoked whenever a pod changed. It checks whether
+// that change made a previously unschedulable pod schedulable.
+func (pl *TaintToleration) isSchedulableAfterPodChange(logger klog.Logger, pod *v1.Pod, oldObj, newObj interface{}) (framework.QueueingHint, error) {
+	originalPod, modifiedPod, err := util.As[*v1.Pod](oldObj, newObj)
+	if err != nil {
+		return framework.Queue, err
+	}
+	if originalPod.Namespace != pod.Namespace ||
+		originalPod.Name != pod.Name ||
+		len(originalPod.Spec.Tolerations) == len(modifiedPod.Spec.Tolerations) {
+		return framework.QueueSkip, nil
+	}
+
+	// We can add, but cannot modify or remove tolerations.
+	// So, it's enough to just check the length of tolerations to notice the update.
+	// And, any updates in tolerations could make Pod schedulable.
+	logger.V(5).Info("a new toleration is added for the Pod, and it may make it schedulable", "pod", klog.KObj(modifiedPod))
+	return framework.Queue, nil
+}
```

Comment:
I think the opposite would be clearer.

```suggestion
	if originalPod.Namespace == pod.Namespace ||
		originalPod.Name == pod.Name ||
		len(originalPod.Spec.Tolerations) != len(modifiedPod.Spec.Tolerations) {
		// An unscheduled Pod got a new toleration.
		// We can add, but cannot modify or remove tolerations.
		// So, it's enough to just check the length of tolerations to notice the update.
		// And, any updates in tolerations could make Pod schedulable.
		logger.V(5).Info("a new toleration is added for the Pod, and it may make it schedulable", "pod", klog.KObj(modifiedPod))
		return framework.Queue, nil
	}

	return framework.QueueSkip, nil
}
```